libraries/AP_Scripting/applets/ScriptingSwitcher/Script_Controller.md

```diff
@@ -1,15 +1,21 @@
 # Script Control LUA Script
 
-This script allows the user to have groups of scripts in three subdirectories of the main scripting directory and copy them into the main directory, removing any files in that directory with the .lua extension (except itself) and issue a scripting restart based on the position of a switch with the auxiliary function of "302". This provides  means to have differing scripting depending on the mission or conditions, and use whichever set is desired without having them consume memory all the time and without requiring a GCS to manage the content of the scripts directory. 
+This script allows the user to have groups of scripts in three subdirectories of the main scripting directory and copy them into the main directory, removing any files in that directory with the .lua extension (except itself) and issue a scripting restart based on the position of a switch with the auxiliary function of "302". This provides  means to have differing scripting depending on the mission or conditions, and use whichever set is desired without having them consume memory all the time and without requiring a GCS to manage the content of the scripts directory.
+
+# Optional Mission and Parameter switching
+
+It is also possible to switch and load a different mission or set of parameters to accompany the new set of scripts at the same time. Simply include a copy of Mission_Controller.lua and/or Param_Controller.lua in each subdirectory that you desire the mission or some parameters changed along with the new scripts to be executed. 
 
 # Setup and Operation
 
-The user must setup an RCx_OPTION to function 302 to determine which of the three scripts subdirectories will be used: LOW:scripts/1, MIDDLE:scripts/2, or HIGH:scripts/3. If no RC has been established during ground start, it will behave as if subdirectory 1 is selected. Changing this switch position either prior to ground start or after, will remove ALL the *.lua files in the scripts directory (except itself), and copy any *.lua files (only) from the desginated subdirectory, and then issue a scripting restart. 
+The user must setup an RCx_OPTION to function 302 to determine which of the three scripts subdirectories will be used: LOW:scripts/1, MIDDLE:scripts/2, or HIGH:scripts/3. If no RC has been established during ground start, it will behave as if subdirectory 1 is selected. Changing this switch position either prior to ground start or after, will remove ALL the *.lua files in the scripts directory (except itself), and copy any *.lua files (only) from the designated subdirectory, and then issue a scripting restart. 
+
+If the mission or parameter script options are used for a subdirectory the mission file to be loaded at the same time that subdirectory's scripts are loaded should be named "mission.txt" and/or "params.param".
```

Comment:
This is a bit awkward. How about:
If mission /or parameter options are  to be loaded at the same time that subdirectory's scripts are loaded, the mission file to be loaded should be named "mission.txt" and the parameter options should be named "params.param".

Reply:
changed wording a bit

Reply:
Not really an improvement - this implies that the mission file can be named mission.txt or params.parm which is definitely not correct. Also it is not a "parameter script", there is no Lua in this file.

Reply:
please reread

Reply:
I stand by my comment and suggestion for improvement.

Reply:
that line was changed to : "If the mission and/or parameter scripts are used and placed in a subdirectory, the mission file to be loaded should be named "mission.txt" and/or the parameter options should be named "params.param" in that sub-directory."  
so your suggestion is outdated....

Reply:
Sorry @Hwurzburg  - it's still showing the old version for me. That new version is much better except drop the first "scripts" (and maybe replace with files?) .. to read.
"If the mission and/or parameter files are used and placed in a subdirectory, the mission file to be loaded should be named "mission.txt" and/or the parameter options should be named "params.param" in that sub-directory."